UseCases/Sources/ManualAsyncHTTPClient/main.swift

```diff
@@ -0,0 +1,109 @@
+import AsyncHTTPClient
+import ContextPropagation
+import NIOHTTP1
+
+let server = FakeHTTPServer(
+    instrumentationMiddlewares: [FakeTracer.Middleware(tracer: FakeTracer())]
+) { context, request, client -> FakeHTTPResponse in
+    print("=== Perform subsequent request ===")
+    let outgoingRequest = try! HTTPClient.Request(
+        url: "https://swift.org",
+        headers: ["Accept": "application/json"]
+    )
+    client.execute(context, request: outgoingRequest)
+//    client.execute(request: outgoingRequest, inContext: context)
+    return FakeHTTPResponse()
+}
+
+print("=== Receive HTTP request on server ===")
+server.receive(try! HTTPClient.Request(url: "https://swift.org"))
+
+// MARK: - InstrumentingHTTPClient
+
+struct InstrumentingHTTPClient {
+    private let client = HTTPClient(eventLoopGroupProvider: .createNew)
+    private let instrumentationMiddlewares: [InstrumentationMiddleware<HTTPHeaders, HTTPHeaders>]
+
+    init<M: InstrumentationMiddlewareProtocol>(
+        instrumentationMiddlewares: [M]
+    ) where M.InjectInto == HTTPHeaders, M.ExtractFrom == HTTPHeaders {
+        self.instrumentationMiddlewares = instrumentationMiddlewares.map {
+            InstrumentationMiddleware(extract: $0.extract, inject: $0.inject)
+        }
+    }
+
+    func execute(request: HTTPClient.Request, inContext context: Context) {
```

Comment:
if we wanted to adopt this style then it should rather be `in context` or `with context`, or `_ context` or `context: ...`...  (Not too hung up on names so far.)

This is a good chance to point people at how the two things read, thanks for pinging Johannes :) 

// more thoughts: https://github.com/slashmo/gsoc-swift-tracing/pull/13#issuecomment-638755592

Reply:
Commented on in https://github.com/slashmo/gsoc-swift-tracing/pull/13#issuecomment-638766188